tests/e2e_tests/conftest.py

```diff
@@ -2,84 +2,99 @@
-        # Skip the test if the localhost.sh path is not set
         logging.warning("LOCALNET_SH_PATH env variable is not set, e2e test skipped.")
         pytest.skip("LOCALNET_SH_PATH environment variable is not set.")
 
-    # Check if param is None, and handle it accordingly
-    args = "" if param is None else f"{param}"
-
-    # Compile commands to send to process
-    cmds = shlex.split(f"{script_path} {args}")
+    # Determine the port to check based on `param`
+    port = 9945  # Default port if `param` is None
 
-    # Start new node process
-    process = subprocess.Popen(
-        cmds, stdout=subprocess.PIPE, text=True, preexec_fn=os.setsid
-    )
-
-    # Pattern match indicates node is compiled and ready
-    pattern = re.compile(r"Imported #1")
-
-    # install neuron templates
-    logging.info("downloading and installing neuron templates from github")
-    # commit with subnet-template-repo changes for rust wallet
+    # Always perform template installation
+    logging.info("Downloading and installing neuron templates from GitHub")
     templates_dir = clone_or_update_templates()
     install_templates(templates_dir)
 
-    timestamp = int(time.time())
-
-    def wait_for_node_start(process, pattern):
-        for line in process.stdout:
-            print(line.strip())
-            # 10 min as timeout
-            if int(time.time()) - timestamp > 10 * 60:
-                print("Subtensor not started in time")
-                break
-            if pattern.search(line):
-                print("Node started!")
-                break
-
-    wait_for_node_start(process, pattern)
-
-    # Run the test, passing in substrate interface
-    yield SubstrateInterface(url="ws://127.0.0.1:9945")
-
-    # Terminate the process group (includes all child processes)
-    os.killpg(os.getpgid(process.pid), signal.SIGTERM)
-
-    # Give some time for the process to terminate
-    time.sleep(1)
-
-    # If the process is not terminated, send SIGKILL
-    if process.poll() is None:
-        os.killpg(os.getpgid(process.pid), signal.SIGKILL)
-
-    # Ensure the process has terminated
-    process.wait()
-
-    # uninstall templates
-    logging.info("uninstalling neuron templates")
-    uninstall_templates(template_path)
+    already_running = False
+    if is_chain_running(port):
+        already_running = True
+        logging.info(f"Chain already running on port {port}, skipping start.")
+    else:
+        logging.info(f"Starting new chain on port {port}...")
+        # compile commands to send to process
+        cmds = shlex.split(f"{script_path} {param}")
+        # Start new node process
+        process = subprocess.Popen(
+            cmds, stdout=subprocess.PIPE, text=True, preexec_fn=os.setsid
+        )
+
+        # Wait for the node to start using the existing pattern match
+        pattern = re.compile(r"Imported #1")
+        timestamp = int(time.time())
+
+        def wait_for_node_start(process, pattern):
```

Comment:
Shadows names `process` and `pattern` from outer scope. pls use names with `_` after var name.